README.md

```diff
@@ -22,12 +27,370 @@ Download the collection and extract it in the `data` directory, under a `collect
+
+The index takes up *66Mo* on the disk.
+
+## Querying
+
+### Loading the index
+
+Since we saved the index as a `pickle` binary the loading is quite fast, around **3 sec**.
+
+### Boolean querying
+
+If the request is entered as *boolean*, we therefore except that it is syntactically correct. We support three logical operator :
+
+- **AND** : `term1 AND term2` will return documents containing both `term1` or `term2`
+- **OR** : `term1 AND term2` will return documents containing `term1`, `term2` or both
+- **NAND** : which means **AND NOT**, `term1 NAND term2` will return documents containing `term1` but not `term2`.
+
+You can also use *parenthesis* for a better query expression. Otherwise, by default, `ttable` has the following default priorities (or *[precedence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence)*) for operators : `NAND > AND > OR`, ie `cats OR dogs NAND ducks AND squid` will be interpreted as `(cats OR ((dogs NAND ducks) AND squid))`
+
+First, we tokenize, remove stop words and lemmatize the query, except for the *logical operators*. Then, we use `ttable` to transform the query into its *postfix syntax*, which will remove parenthesis and change the order of tokens and logical operator. It is then easy to process the query : read the postfix query sequentially. If you have a token, add the list of documents with this token (with the inverted index). Ortherwise, its a logical operator, apply a merge with this operator and the last two elements of the stack.
+
+If you don't specify any logical operator, we assume it's an **AND** request : `cats dog duck` --> `cats AND dog AND duck`.
+
+### Vectorial querying
+
+Here we're making a dot product between a vector representing the query and the vector representing the dataset.
+
+**TODO**: add a few more lines
+
+### Practical querying
+
+To use and execute queries, you can use the cli provided by `[src/interface.py](src/interface.py)`.
+
+```bash
+> python src/interface.py --help
+usage: interface.py [-h] [--model MODEL] [--number NUMBER] query
+
+positional arguments:
+  query                 query to process
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --model MODEL         <boolean|vectorial> model used to process query (default="boolean")
+  --number NUMBER, -n NUMBER
+                        number of results to display (default=10)
+```
+
+The path to the index or the stats used for vectorial query are configured through *Environnement variable*, which can also be modified in `[src/config.py](src/config.py)`.
+
+For example, if you want to use a custom index, for a vectorial query with a `tf_idf_log_normalize` for both query and document weights, you could run the following query :
+
+```bash
+PATH_INDEX=/my/custom/index/path WEIGHT_QUERY=tf_idf_log_normalize WEIGHT_DOCUMENT=tf_idf_log_normalize python src/interface.py --model vectorial "cats are cute"
```

Comment:
It's a bit ugly to mix cli parameters and env variables. You can add optional cli parameters.